django/app_tutorial/models.py

```diff
@@ -4,43 +4,80 @@
 from django.contrib.auth.models import User
 
 
-class Nationality(models.Model):
-    abbreviation = models.CharField(max_length=5)
-    name = models.CharField(max_length=75)
+class DivisionAge(models.Model):
+    name = models.CharField(max_length=50)
+    # In our discussion this was denoted as type, I thought this might be handier.
+    age_min = models.IntegerField()
+    age_max = models.IntegerField()
 
 
-class Division(models.Model):
+class DivisionType(models.Model):
+    # TODO (Feedback): In our discussion this was division gender, but I feel like Mixed and Open aren't really genders.
     name = models.CharField(max_length=50)
 
 
-class TournamentType(models.Model):
-    name = models.CharField(max_length=50)
+class FieldType(models.Model):
+    type = models.CharField(max_length=50)
 
 
-class GameType(models.Model):
-    name = models.CharField(max_length=50)
+class Field(models.Model):
+    location = models.CharField(max_length=100)
+    type = models.ForeignKey(FieldType)
 
 
-class GameReportEventType(models.Model):
+class Location(models.Model):
     name = models.CharField(max_length=50)
+    location = models.CharField(max_length=100)
 
 
-class SpiritScoreCategory(models.Model):
-    name = models.CharField(max_length=50)
+class Tournament(models.Model):
+    name = models.CharField(max_length=75)
+    location = models.ForeignKey(Location)
+    date_begin = models.DateField('Begin Date')
+    date_end = models.DateField('End Date')
+    # type = models.ForeignKey(TournamentType)
 
 
-class Team(models.Model):
+class TournamentDivision(models.Model):
+    tournament = models.ForeignKey(Tournament)
+    age_group = models.ForeignKey(DivisionAge)
+    division = models.ForeignKey(DivisionType)
+
+
+class TournamentField(models.Model):
+    tournament = models.ForeignKey(Tournament)
+    field = models.ForeignKey(Field)
+    field_number = models.IntegerField()  # TODO (Feedback): imo this is way more practical than a string name
+
+
+class Squad(models.Model):
+    # TODO: When teams are added, make name nullable and build squad name from division age and type if name is empty.
+    division_age = models.ForeignKey(DivisionAge)
+    division_type = models.ForeignKey(DivisionType)
     name = models.CharField(max_length=50)
-    origin = models.CharField(max_length=50)
-    division = models.ForeignKey(Division)
+    # team = models.ForeignKey(Team)
+
+
+class Game(models.Model):
+    division = models.ForeignKey(TournamentDivision)  # TODO: Redundant? Contained in Squad.division
+    date = models.DateTimeField('Game Time')
+    squad_home = models.ForeignKey(Squad, related_name='squad_home')  # TODO: What does related_name do?
```

Comment:
IMHO it is not relevant for the data base model at all. Rather, this seems Django specific as Django creates custom relationship methods on model.Model instances which contain relations to each other.

For a minimal example, consider the following **models.py**:
```python
from django.db import models

class Blog(models.Model):
    name = models.CharField(max_length=100)

class Entry(models.Model):
    blog = models.ForeignKey(Blog) 

```

Lets say we've already saved several blog and entry objects to our data base. Django allows you to access the blog of an entry (forward) and the entries of an blog (backwards) because they are in a one-to-many relationship due to the ForeignKey:

```python

from models import Blog, Entry

# Forward
e = Entry.objects.get(id=1) # get entry object
e.blog # Returns the related Blog object.

# Backwards
b = Blog.objects.get(id=1) # get blog object
b.entry_set.all() # Returns all Entry objects related to Blog.

```
Now, if you have two foreign keys relating to the same table (Game table having two FKs to Squad table), Django basically can't create those *relationship-accessor-methods* properly because the name of the referencing table (Game) would be used twice. Therefore, the `related_name` allows you to specify a custom name to be used instead.

To be more concrete:
```python
from models import Squad

s = Squad.objects.get(id=1) # get first squad
s.squad_home_set().all() # get all Game objects, where squad was home squad
s.squad_away_set().all() # get all Game objects, where squad was away squad
```

I'm not entirely sure about this but see this [backwards-related-objects](https://docs.djangoproject.com/ja/1.10/topics/db/queries/#backwards-related-objects)